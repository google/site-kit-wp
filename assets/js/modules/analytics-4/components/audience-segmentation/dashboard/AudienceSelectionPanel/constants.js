/**
 * AudienceSelectionPanel components - constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const AUDIENCE_ADD_GROUP_NOTICE_SLUG =
	'audience-segmentation-add-group-notice';
export const AUDIENCE_SELECTION_PANEL_OPENED_KEY =
	'googlesitekit-audience-selection-panel-opened';
export const AUDIENCE_SELECTION_FORM = 'audience-selection-form';
export const AUDIENCE_SELECTED = 'audience-selected';
export const AUDIENCE_SELECTION_CHANGED = 'audience-selection-changed';
export const AUDIENCE_CREATION_NOTICE_SLUG =
	'audience-segmentation-creation-notice';
export const AUDIENCE_CREATION_SUCCESS_NOTICE_SLUG =
	'audience-segmentation-creation-success-notice';
export const MIN_SELECTED_AUDIENCES_COUNT = 1;
export const MAX_SELECTED_AUDIENCES_COUNT = 3;
export const AUDIENCE_CREATION_EDIT_SCOPE_NOTICE_SLUG =
	'audience-creation-edit-scope-notice';
export const AUDIENCE_CREATION_FORM = 'audience-creation-form';
