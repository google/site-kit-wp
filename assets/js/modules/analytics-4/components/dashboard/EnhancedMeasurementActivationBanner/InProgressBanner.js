/**
 * EnhancedMeasurementActivationBanner > InProgressBanner component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useSelect } from 'googlesitekit-data';
import { CORE_SITE } from '../../../../../googlesitekit/datastore/site/constants';
import SetupCTA from '../../../../../googlesitekit/notifications/components/layout/SetupCTA';
import BannerSVG from '@/svg/graphics/banner-enhanced-measurement-setup-cta.svg?url';
import BannerMobileSVG from '@/svg/graphics/banner-enhanced-measurement-setup-cta-mobile.svg?url';
import { CORE_USER } from '@/js/googlesitekit/datastore/user/constants';
import { EDIT_SCOPE } from '../../../datastore/constants';

export default function InProgressBanner( { id, Notification, onDismiss } ) {
	const documentationURL = useSelect( ( select ) =>
		select( CORE_SITE ).getGoogleSupportURL( {
			path: '/analytics/answer/9216061',
		} )
	);

	const hasEditScope = useSelect( ( select ) =>
		select( CORE_USER ).hasScope( EDIT_SCOPE )
	);

	const description = hasEditScope
		? __(
				'Enable enhanced measurement in Analytics to automatically track metrics like file downloads, video plays, form interactions, etc. No extra code required.',
				'google-site-kit'
		  )
		: __(
				'Enable enhanced measurement in Analytics to automatically track metrics like file downloads, video plays, form interactions, etc. No extra code required — you’ll be redirected to give permission for Site Kit to enable it on your behalf.',
				'google-site-kit'
		  );

	return (
		<Notification>
			<SetupCTA
				notificationID={ id }
				title={ __(
					'Understand how visitors interact with your content',
					'google-site-kit'
				) }
				description={ description }
				learnMoreLink={ {
					href: documentationURL,
				} }
				ctaButton={ {
					label: __( 'Enable now', 'google-site-kit' ),
					disabled: true,
				} }
				dismissButton={ {
					label: __( 'Maybe later', 'google-site-kit' ),
					onClick: onDismiss,
				} }
				waitingProgress={ {
					height: 7,
					indeterminate: true,
				} }
				helpText={ __(
					'You can always add/edit this in the Site Kit Settings',
					'google-site-kit'
				) }
				svg={ {
					desktop: BannerSVG,
					mobile: BannerMobileSVG,
					verticalPosition: 'center',
				} }
			/>
		</Notification>
	);
}

InProgressBanner.propTypes = {
	id: PropTypes.string.isRequired,
	Notification: PropTypes.elementType,
	onDismiss: PropTypes.func,
};
