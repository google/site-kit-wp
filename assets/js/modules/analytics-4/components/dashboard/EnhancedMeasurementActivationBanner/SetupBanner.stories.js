/**
 * EnhancedMeasurementActivationBanner > SetupBanner Component Stories.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { provideUserAuthentication } from '../../../../../../../tests/js/utils';
import WithRegistrySetup from '../../../../../../../tests/js/WithRegistrySetup';
import { withNotificationComponentProps } from '../../../../../googlesitekit/notifications/util/component-props';
import { EDIT_SCOPE } from '../../../datastore/constants';
import SetupBanner from './SetupBanner';

const NotificationWithComponentProps = withNotificationComponentProps(
	'enhanced-measurement-notification'
)( SetupBanner );

function Template() {
	return <NotificationWithComponentProps />;
}

export const Default = Template.bind( {} );
Default.storyName = 'Default';
Default.args = {
	grantedScopes: [ EDIT_SCOPE ],
};
Default.scenario = {};

export const NoEditScope = Template.bind( {} );
NoEditScope.storyName = 'No Edit Scope';

export default {
	title: 'Modules/Analytics4/EnhancedMeasurementActivationBanner/SetupBanner',
	decorators: [
		( Story, { args: { grantedScopes } } ) => {
			function setupRegistry( registry ) {
				provideUserAuthentication( registry, {
					grantedScopes,
				} );
			}

			return (
				<WithRegistrySetup func={ setupRegistry }>
					<Story />
				</WithRegistrySetup>
			);
		},
	],
};
