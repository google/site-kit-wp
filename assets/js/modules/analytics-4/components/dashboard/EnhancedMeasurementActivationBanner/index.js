/**
 * EnhancedMeasurementActivationBanner component.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { useCallback, useEffect, useState } from '@wordpress/element';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useSelect, useDispatch } from 'googlesitekit-data';
import { CORE_FORMS } from '../../../../../googlesitekit/datastore/forms/constants';
import { CORE_MODULES } from '../../../../../googlesitekit/modules/datastore/constants';
import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
import {
	EDIT_SCOPE,
	ENHANCED_MEASUREMENT_ENABLED,
	ENHANCED_MEASUREMENT_FORM,
	FORM_SETUP,
	MODULES_ANALYTICS_4,
} from '../../../datastore/constants';
import {
	ACTIVATION_STEP_IN_PROGRESS,
	ACTIVATION_STEP_SETUP,
	ACTIVATION_STEP_SUCCESS,
	ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY,
	ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY,
} from '../../../constants';
import { useTooltipState } from '../../../../../components/AdminMenuTooltip/useTooltipState';
import { useShowTooltip } from '../../../../../components/AdminMenuTooltip/useShowTooltip';
import { AdminMenuTooltip } from '../../../../../components/AdminMenuTooltip/AdminMenuTooltip';
import InProgressBanner from './InProgressBanner';
import SetupBanner from './SetupBanner';
import SuccessBanner from './SuccessBanner';
import { MONTH_IN_SECONDS, trackEvent } from '../../../../../util';
import whenActive from '../../../../../util/when-active';
import {
	isValidPropertyID,
	isValidWebDataStreamID,
} from '../../../utils/validation';
import useViewContext from '../../../../../hooks/useViewContext';

function EnhancedMeasurementActivationBanner() {
	const viewContext = useViewContext();

	const [ step, setStep ] = useState( ACTIVATION_STEP_SETUP );
	const [
		isEnhancedMeasurementInitiallyDisabled,
		setIsEnhancedMeasurementInitiallyDisabled,
	] = useState( undefined );
	const [ isSaving, setIsSaving ] = useState( false );
	const [ errorNotice, setErrorNotice ] = useState( null );

	const propertyID = useSelect( ( select ) =>
		select( MODULES_ANALYTICS_4 ).getPropertyID()
	);

	const webDataStreamID = useSelect( ( select ) =>
		select( MODULES_ANALYTICS_4 ).getWebDataStreamID()
	);

	const isBannerDismissed = useSelect( ( select ) =>
		select( CORE_USER ).isItemDismissed(
			ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY
		)
	);

	const isDismissingBanner = useSelect( ( select ) =>
		select( CORE_USER ).isDismissingItem(
			ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY
		)
	);

	const hasModuleAccess = useSelect( ( select ) =>
		select( CORE_MODULES ).hasModuleOwnershipOrAccess( 'analytics-4' )
	);

	const isEnhancedMeasurementStreamEnabled = useSelect( ( select ) => {
		if (
			! isValidPropertyID( propertyID ) ||
			! isValidWebDataStreamID( webDataStreamID ) ||
			! hasModuleAccess ||
			isBannerDismissed
		) {
			return undefined;
		}

		return select( MODULES_ANALYTICS_4 ).isEnhancedMeasurementStreamEnabled(
			propertyID,
			webDataStreamID
		);
	} );

	const hasEditScope = useSelect( ( select ) =>
		select( CORE_USER ).hasScope( EDIT_SCOPE )
	);

	const autoSubmit = useSelect( ( select ) =>
		select( CORE_FORMS ).getValue( FORM_SETUP, 'autoSubmit' )
	);

	const { setValues } = useDispatch( CORE_FORMS );
	const { dismissItem } = useDispatch( CORE_USER );
	const { submitChanges } = useDispatch( MODULES_ANALYTICS_4 );

	const { isTooltipVisible } = useTooltipState(
		ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY
	);

	const showTooltip = useShowTooltip(
		ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY
	);

	function handleDismiss() {
		showTooltip();
		dismissItem(
			ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY,
			{
				expiresInSeconds: MONTH_IN_SECONDS,
			}
		);
	}

	const handleSubmit = useCallback( async () => {
		setIsSaving( true );

		setValues( ENHANCED_MEASUREMENT_FORM, {
			[ ENHANCED_MEASUREMENT_ENABLED ]: true,
		} );

		const { error } = await submitChanges();

		setIsSaving( false );

		if ( error ) {
			setErrorNotice( error );
			setStep( ACTIVATION_STEP_SETUP );
			return;
		}

		trackEvent(
			`${ viewContext }_enhanced-measurement-notification`,
			'confirm_notification'
		);

		setStep( ACTIVATION_STEP_SUCCESS );
	}, [ setValues, submitChanges, viewContext ] );

	useEffect( () => {
		if (
			isEnhancedMeasurementStreamEnabled === false &&
			isEnhancedMeasurementInitiallyDisabled === undefined
		) {
			setIsEnhancedMeasurementInitiallyDisabled( true );
		}
	}, [
		isEnhancedMeasurementInitiallyDisabled,
		isEnhancedMeasurementStreamEnabled,
	] );

	// If the user lands back on this component with autoSubmit and the edit scope,
	// resubmit the form.
	useEffect( () => {
		async function handleAutoSubmit() {
			// Auto-submit should only auto-invoke once.
			setValues( FORM_SETUP, { autoSubmit: false } );

			setStep( ACTIVATION_STEP_IN_PROGRESS );

			await handleSubmit();
		}

		if ( autoSubmit && hasEditScope ) {
			handleAutoSubmit();
		}
	}, [ hasEditScope, setValues, handleSubmit, autoSubmit ] );

	if ( isTooltipVisible ) {
		return (
			<AdminMenuTooltip
				title={ __(
					'Enable enhanced measurement later here',
					'google-site-kit'
				) }
				content={ __(
					'You can always turn on enhanced measurement later in Site Kit Settings',
					'google-site-kit'
				) }
				dismissLabel={ __( 'Got it', 'google-site-kit' ) }
				tooltipStateKey={
					ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY
				}
			/>
		);
	}

	if (
		! isEnhancedMeasurementInitiallyDisabled ||
		isBannerDismissed ||
		isDismissingBanner
	) {
		return null;
	}

	if ( step === ACTIVATION_STEP_SETUP ) {
		return (
			<SetupBanner
				errorNotice={ errorNotice }
				isSaving={ isSaving }
				onDismiss={ handleDismiss }
				onSubmit={ handleSubmit }
			/>
		);
	}

	if ( step === ACTIVATION_STEP_IN_PROGRESS ) {
		return <InProgressBanner />;
	}

	if ( step === ACTIVATION_STEP_SUCCESS ) {
		return <SuccessBanner />;
	}

	return null;
}

export default whenActive( { moduleName: 'analytics-4' } )(
	EnhancedMeasurementActivationBanner
);
