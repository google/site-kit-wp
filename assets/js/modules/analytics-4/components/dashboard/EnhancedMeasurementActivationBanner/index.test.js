/**
 * EnhancedMeasurementActivationBanner component tests.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import {
	createTestRegistry,
	render,
	provideModules,
	provideModuleRegistrations,
	provideUserInfo,
	provideUserAuthentication,
	fireEvent,
	waitFor,
	waitForElementToBeRemoved,
	freezeFetch,
} from '../../../../../../../tests/js/test-utils';
import { CORE_FORMS } from '../../../../../googlesitekit/datastore/forms/constants';
import { CORE_MODULES } from '../../../../../googlesitekit/modules/datastore/constants';
import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
import {
	EDIT_SCOPE,
	FORM_SETUP,
	MODULES_ANALYTICS_4,
} from '../../../datastore/constants';
import { ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY } from '../../../constants';
import * as analytics4Fixtures from '../../../datastore/__fixtures__';
import EnhancedMeasurementActivationBanner from './index';
import { properties } from '../../../datastore/__fixtures__';

describe( 'EnhancedMeasurementActivationBanner', () => {
	const propertyID = '1000';
	const webDataStreamID = '2000';

	const enhancedMeasurementSettingsEndpoint = new RegExp(
		'^/google-site-kit/v1/modules/analytics-4/data/enhanced-measurement-settings'
	);

	let enhancedMeasurementSettingsMock;
	let enhancedMeasurementEnabledSettingsMock;
	let registry;

	beforeEach( () => {
		enhancedMeasurementSettingsMock = {
			fileDownloadsEnabled: null,
			name: 'properties/1000/dataStreams/2000/enhancedMeasurementSettings',
			outboundClicksEnabled: null,
			pageChangesEnabled: null,
			scrollsEnabled: null,
			searchQueryParameter: 'q,s,search,query,keyword',
			siteSearchEnabled: null,
			streamEnabled: false,
			uriQueryParameter: null,
			videoEngagementEnabled: null,
		};

		enhancedMeasurementEnabledSettingsMock = {
			...enhancedMeasurementSettingsMock,
			streamEnabled: true,
		};

		registry = createTestRegistry();

		provideUserInfo( registry );
		provideUserAuthentication( registry, {
			grantedScopes: [ EDIT_SCOPE ],
		} );
		provideModules( registry, [
			{
				slug: 'analytics-4',
				active: true,
				connected: true,
			},
		] );
		provideModuleRegistrations( registry );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
			propertyID,
			webDataStreamID,
			ownerID: 1,
			propertyCreateTime: 1662715085968,
		} );

		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );

		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.receiveGetProperty( properties[ 0 ], { propertyID } );

		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.receiveGetEnhancedMeasurementSettings(
				enhancedMeasurementSettingsMock,
				{ propertyID, webDataStreamID }
			);
	} );

	it( 'should render the setup step when enhanced measurement is initially false and the banner is not dismissed', async () => {
		const { container, getByRole } = render(
			<EnhancedMeasurementActivationBanner />,
			{
				registry,
			}
		);

		await waitFor( () => expect( container ).toMatchSnapshot() );

		await waitFor( () =>
			expect(
				getByRole( 'button', { name: 'Enable now' } )
			).toBeInTheDocument()
		);
	} );

	it( 'should render the in progress step when enhanced measurement is being enabled after the user returns from the OAuth flow', async () => {
		freezeFetch( enhancedMeasurementSettingsEndpoint );

		registry
			.dispatch( CORE_FORMS )
			.setValues( FORM_SETUP, { autoSubmit: true } );

		const { container, getByText } = render(
			<EnhancedMeasurementActivationBanner />,
			{
				registry,
			}
		);

		await waitFor( () => expect( container ).toMatchSnapshot() );

		expect( getByText( 'Setup in progress' ) ).toBeInTheDocument();
	} );

	it( 'should render the success step when the the setup form is successfully submitted', async () => {
		fetchMock.postOnce( enhancedMeasurementSettingsEndpoint, {
			status: 200,
			body: analytics4Fixtures.defaultEnhancedMeasurementSettings,
		} );

		const { container, getByRole, getByText, waitForRegistry } = render(
			<EnhancedMeasurementActivationBanner />,
			{
				registry,
			}
		);

		await waitForRegistry();

		const enableNowButton = getByRole( 'button', { name: 'Enable now' } );
		fireEvent.click( enableNowButton );

		await waitForElementToBeRemoved( enableNowButton );

		await waitForRegistry();

		// Enhanced measurement settings should update when enhanced measurement
		// is enabled via the "Enable now" CTA.
		expect( fetchMock ).toHaveBeenCalledTimes(
			1,
			enhancedMeasurementSettingsEndpoint
		);

		expect( container ).toMatchSnapshot();

		expect(
			getByText(
				'You successfully enabled enhanced measurement for your site'
			)
		).toBeInTheDocument();
	} );

	it.each( [
		[
			'there is not a valid propertyID',
			() => {
				registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
					propertyID: '',
				} );
			},
		],
		[
			'there is not a valid webDataStreamID',
			() => {
				registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
					webDataStreamID: '',
				} );
			},
		],
		[
			'the user does not have access to the Analytics 4 module',
			() => {
				registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
					ownerID: 2,
				} );

				registry
					.dispatch( CORE_MODULES )
					.receiveCheckModuleAccess(
						{ access: false },
						{ slug: 'analytics-4' }
					);
			},
		],
		[
			'enhanced measurement is initially true',
			() => {
				registry
					.dispatch( MODULES_ANALYTICS_4 )
					.receiveGetEnhancedMeasurementSettings(
						{
							...enhancedMeasurementSettingsMock,
							streamEnabled: true,
						},
						{ propertyID, webDataStreamID }
					);
			},
		],
		[
			'the banner is dismissed',
			() => {
				registry
					.dispatch( CORE_USER )
					.receiveGetDismissedItems( [
						ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY,
					] );
			},
		],
	] )( 'should not render when %s', ( _, setupTestCase ) => {
		setupTestCase();

		const { container } = render( <EnhancedMeasurementActivationBanner />, {
			registry,
		} );

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should enable enhanced measurement when the CTA in SetupBanner is clicked and the user has the edit scope granted', async () => {
		const { getByRole, waitForRegistry } = render(
			<EnhancedMeasurementActivationBanner />,
			{
				registry,
			}
		);

		fetchMock.postOnce( enhancedMeasurementSettingsEndpoint, {
			status: 200,
			body: { ...enhancedMeasurementEnabledSettingsMock },
		} );

		fireEvent.click( getByRole( 'button', { name: 'Enable now' } ) );

		await waitForRegistry();

		expect( fetchMock ).toHaveFetched(
			enhancedMeasurementSettingsEndpoint,
			{
				body: {
					data: {
						propertyID,
						webDataStreamID,
						enhancedMeasurementSettings:
							enhancedMeasurementEnabledSettingsMock,
					},
				},
			}
		);
	} );

	it( 'should enable enhanced measurement when the form is auto submitted after the edit scope was granted', async () => {
		registry
			.dispatch( CORE_FORMS )
			.setValues( FORM_SETUP, { autoSubmit: true } );

		fetchMock.postOnce( enhancedMeasurementSettingsEndpoint, {
			status: 200,
			body: { ...enhancedMeasurementEnabledSettingsMock },
		} );

		const { waitForRegistry } = render(
			<EnhancedMeasurementActivationBanner />,
			{
				registry,
			}
		);

		await waitForRegistry();

		expect( fetchMock ).toHaveFetched(
			enhancedMeasurementSettingsEndpoint,
			{
				body: {
					data: {
						propertyID,
						webDataStreamID,
						enhancedMeasurementSettings:
							enhancedMeasurementEnabledSettingsMock,
					},
				},
			}
		);
	} );

	it( 'should not render the banner when the prompt is being dismissed', async () => {
		registry
			.dispatch( CORE_USER )
			.setIsItemDimissing(
				ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY,
				true
			);

		const { container, waitForRegistry } = render(
			<EnhancedMeasurementActivationBanner />,
			{
				registry,
			}
		);

		await waitForRegistry();

		expect( container ).toBeEmptyDOMElement();
	} );
} );
