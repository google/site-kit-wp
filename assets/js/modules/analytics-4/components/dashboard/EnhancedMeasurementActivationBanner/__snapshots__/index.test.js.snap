// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnhancedMeasurementActivationBanner should render the in progress step when enhanced measurement is being enabled after the user returns from the OAuth flow 1`] = `
<div>
  <section
    class="googlesitekit-publisher-win googlesitekit-enhanced-measurement-success-banner"
    id="enhanced-measurement-notification"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-publisher-win__content mdc-layout-grid__cell mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-6-tablet mdc-layout-grid__cell--span-4-phone"
        >
          <h3
            class="googlesitekit-heading-2 googlesitekit-publisher-win__title"
          >
            Setup in progress
          </h3>
          <div
            class="googlesitekit-publisher-win__desc"
          >
            <p>
              <span>
                Enhanced measurement is being enabled.
              </span>
               
              <a
                aria-label="Learn more (opens in a new tab)"
                class="googlesitekit-cta-link"
                href="https://support.google.com/analytics/answer/9216061?hl=en-US"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Learn more
                </span>
                <span
                  class="googlesitekit-icon-wrapper"
                  style="margin-left: 5px;"
                >
                  <svg />
                </span>
              </a>
            </p>
          </div>
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 20px; height: 20px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`EnhancedMeasurementActivationBanner should render the setup step when enhanced measurement is initially false and the banner is not dismissed 1`] = `
<div>
  <section
    class="googlesitekit-publisher-win googlesitekit-enhanced-measurement-setup-banner"
    id="enhanced-measurement-notification"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-publisher-win__content mdc-layout-grid__cell mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-6-tablet mdc-layout-grid__cell--span-4-phone"
        >
          <h3
            class="googlesitekit-heading-2 googlesitekit-publisher-win__title"
          >
            Understand how visitors interact with your content
          </h3>
          <div
            class="googlesitekit-publisher-win__desc"
          >
            <p>
              <span>
                Enable enhanced measurement in Analytics to automatically track metrics like file downloads, video plays, form interactions, etc. No extra code required.
              </span>
               
              <a
                aria-label="Learn more (opens in a new tab)"
                class="googlesitekit-cta-link"
                href="https://support.google.com/analytics/answer/9216061?hl=en-US"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Learn more
                </span>
                <span
                  class="googlesitekit-icon-wrapper"
                  style="margin-left: 5px;"
                >
                  <svg />
                </span>
              </a>
            </p>
          </div>
          <div
            class="googlesitekit-publisher-win__actions"
          >
            <button
              class="mdc-button googlesitekit-notification__cta googlesitekit-button-icon--spinner googlesitekit-button-icon--spinner__after mdc-button--raised"
              target="_self"
            >
              <span
                class="mdc-button__label"
              >
                Enable now
              </span>
            </button>
            <button
              class="mdc-button mdc-button--tertiary"
              target="_self"
            >
              <span
                class="mdc-button__label"
              >
                Maybe later
              </span>
            </button>
          </div>
          <div
            class="googlesitekit-publisher-win__footer"
          >
            <p
              class="googlesitekit-enhanced-measurement-setup-banner__footer-notice"
            >
              You can always add/edit this in the Site Kit Settings
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`EnhancedMeasurementActivationBanner should render the success step when the the setup form is successfully submitted 1`] = `
<div>
  <section
    class="googlesitekit-publisher-win googlesitekit-enhanced-measurement-success-banner"
    id="enhanced-measurement-notification"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-publisher-win__content mdc-layout-grid__cell mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-6-tablet mdc-layout-grid__cell--span-4-phone"
        >
          <h3
            class="googlesitekit-heading-2 googlesitekit-publisher-win__title"
          >
            You successfully enabled enhanced measurement for your site
          </h3>
          <div
            class="googlesitekit-publisher-win__desc"
          >
            <p>
              <span>
                Your configured Analytics web data stream will now automatically measure interactions on your site in addition to standard page views measurement.
              </span>
               
              <a
                aria-label="Learn more (opens in a new tab)"
                class="googlesitekit-cta-link"
                href="https://support.google.com/analytics/answer/9216061?hl=en-US"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Learn more
                </span>
                <span
                  class="googlesitekit-icon-wrapper"
                  style="margin-left: 5px;"
                >
                  <svg />
                </span>
              </a>
            </p>
          </div>
          <button
            class="mdc-button googlesitekit-notification__cta googlesitekit-button-icon--spinner googlesitekit-button-icon--spinner__after mdc-button--raised"
            target="_self"
          >
            <span
              class="mdc-button__label"
            >
              OK, Got it
            </span>
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`EnhancedMeasurementActivationBanner should render the success step when the the setup form is successfully submitted 2`] = `
<div>
  <section
    class="googlesitekit-enhanced-measurement-success-banner googlesitekit-publisher-win--small googlesitekit-publisher-win--win-success googlesitekit-publisher-win--is-open googlesitekit-publisher-win"
    id="googlesitekit-enhanced-measurement-activation-banner"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-publisher-win__content mdc-layout-grid__cell mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-6-tablet mdc-layout-grid__cell--span-4-phone"
        >
          <div
            class="googlesitekit-publisher-win__title-image-wrapper"
          >
            <h3
              class="googlesitekit-heading-2 googlesitekit-publisher-win__title"
            >
              You successfully enabled enhanced measurement for your site
            </h3>
            <div
              class="googlesitekit-publisher-win__image-small"
            >
              <svg />
            </div>
          </div>
          <div
            class="googlesitekit-publisher-win__desc"
          >
            <p>
              <span>
                Your configured Analytics web data stream will now automatically measure interactions on your site in addition to standard page views measurement.
              </span>
               
              <a
                aria-label="Learn more (opens in a new tab)"
                class="googlesitekit-cta-link"
                href="https://support.google.com/analytics/answer/9216061?hl=en-US"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="googlesitekit-cta-link__contents"
                >
                  Learn more
                </span>
                <span
                  class="googlesitekit-icon-wrapper"
                  style="margin-left: 5px;"
                >
                  <svg />
                </span>
              </a>
            </p>
          </div>
          <div
            class="googlesitekit-publisher-win__actions"
          >
            <button
              class="mdc-button mdc-button--raised"
              target="_self"
            >
              <span
                class="mdc-button__label"
              >
                OK, Got it
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
